File: Backend/Entities/StaffTraining.cs
using System;

namespace Backend.Entities
{
    public class StaffTraining : BaseEntity
    {
        public Guid StaffId { get; set; }
        public Guid TrainingRequirementId { get; set; }
        public DateTime? CompletedDate { get; set; }
    }

    public class StaffTrainingWithRequirement : StaffTraining
    {
        public string RequirementName { get; set; }
        public TrainingScope RequirementScope { get; set; }
    }
}

File: Frontend/src/app/people/training-requirement/staff-training.ts
export class StaffTraining {
  public id: string;
  public staffId: string;
  public trainingRequirementId: string;
  public completedDate: Date;

  static getKey(staffTraining: StaffTraining) {
    return staffTraining.staffId + '_' + staffTraining.trainingRequirementId;
  }

  static New(staffId: string, trainingId?: string) {
    let training = new StaffTraining();
    training.staffId = staffId;
    training.trainingRequirementId = trainingId;
    return training;
  }
}

export class StaffTrainingWithRequirement extends StaffTraining {
  public requirementName: string;
  public requirementScope: string;
}
